contracts/UpgradeableAssetToken.sol

```diff
@@ -42,10 +42,10 @@ contract UpgradeableAssetToken is Initializable, ERC223Interface, IERC20 {
         for every new update - as a constructor
     */
 
-    function initialize(string memory symbol, string memory name) public initializer {
+    function initialize(string memory symbol, string memory name, address owner) public initializer {
         _symbol = symbol;
         _name = name;
-        _owner = msg.sender;
+        _owner = owner;
```

Comment:
Being public and callable by anyone, this could be a security hole. An attacker could initialise the contract with their own values and take control